explorer/test/Test/Pos/Explorer/Socket/MethodsSpec.hs

```diff
@@ -68,3 +122,191 @@ addressSetByTxsProp key =
         addrs = addressSetByTxs txNE txsNE
     in
         addrs == S.fromList [addrA]
+
+subscribeAddrProp :: Property
+subscribeAddrProp =
+    forAll arbitrary $ \(socketId, addr) ->
+        monadicIO $ do
+            let connState = mkSubConnectionState socketId
+            let cAddr = toCAddress addr
+            let subscription = runSubTestMode connState $
+                                  subscribeAddr cAddr socketId
+
+            (_, updatedConnState) <- run subscription
+
+            -- get sessions `Address` subscribers
+            let mSessions = updatedConnState ^. csAddressSubscribers . at addr
+            -- to check whether current session has been added to it or not
+            assert $ hasSession socketId mSessions
+  where
+    hasSession :: SocketId -> Maybe (S.Set SocketId) -> Bool
+    hasSession socketId' (Just sessions) = S.member socketId' sessions
+    hasSession _          Nothing        = False
+
+unsubscribeAddrProp :: Property
+unsubscribeAddrProp =
+    forAll arbitrary $ \(socketId, addr) ->
+        monadicIO $ do
+            let cAddr = toCAddress addr
+            let connState = mkSubConnectionState socketId
+            -- Add a subscription first ...
+            let subscription = runSubTestMode connState $
+                                  subscribeAddr cAddr socketId
+            (_, updatedConnState) <- run subscription
+            -- ... and remove subscription
```

Comment:
You could `assert` in the middle of the tests, since it could simply not work and the end result would be `M.size sessions == 0`?

Reply:
As discussed in Slack w/ Kristijan and me: Extra `assert` would compare same things as we already have with `subscribe...` tests. So there is no need for doing it here.